echo_server/lib/echo_server.ex

```diff
@@ -0,0 +1,42 @@
+defmodule EchoServer do
+  use GenServer
+
+  @moduledoc """
+  Documentation for `EchoServer`.
+  """
+
+  @doc """
+  Ping-pong.
+
+  ## Examples
+      iex> {:ok, pid} = EchoServer.start_link()
+      iex> EchoServer.echo(pid, :ping)
+      {:pong, :nonode@nohost}
+
+  """
+
+  # Client
+
+  def start_link(default \\ %{}) do
+    GenServer.start_link(__MODULE__, default)
+  end
+
+  def echo(pid, message) do
+    case message do
+      :ping -> GenServer.call(pid, message)
+      _ -> raise "Only :ping message is allowed!"
+    end
+  end
```

Comment:
Не нужно переписывать своими словами корку эликсира :)
```suggestion
  def echo(pid, :ping) do
    GenServer.call(pid, message)
  end
```